File: app/src/main/java/com/esafirm/androidplayground/reductor/CounterActions.java
package com.esafirm.androidplayground.reductor;

import com.yheriatovych.reductor.Action;
import com.yheriatovych.reductor.annotations.ActionCreator;

@ActionCreator
public interface CounterActions {
    String INCREMENT = "INCREMENT";
    String ADD = "ADD";

    @ActionCreator.Action(INCREMENT)
    Action increment();

    @ActionCreator.Action(ADD)
    Action add(int value);
}

File: app/src/main/java/com/esafirm/androidplayground/reductor/CounterActions.kt
package com.esafirm.androidplayground.reductor

import com.yheriatovych.reductor.Action
import com.yheriatovych.reductor.annotations.ActionCreator

@ActionCreator
interface CounterActions {

    @ActionCreator.Action(INCREMENT)
    fun increment(): Action

    @ActionCreator.Action(ADD)
    fun add(value: Int): Action

    companion object {
        const val INCREMENT = "INCREMENT"
        const val ADD = "ADD"
    }
}
